python/hsfs/core/transformation_function_api.py

```diff
@@ -0,0 +1,117 @@
+#
+
+from hsfs import client, transformation_function, transformation_function_attached
+
+
+class TransformationFunctionApi:
+    def __init__(self, feature_store_id):
+        self._feature_store_id = feature_store_id
+
+    def register_transformation_fn(self, transformation_function_instance):
+        """
+        Register transformation function in backend
+        Args:
+        transformation_function_instance: TransformationFunction, required
+            metadata object of transformation function.
+        """
+        _client = client.get_instance()
+        path_params = [
+            "project",
+            _client._project_id,
+            "featurestores",
+            self._feature_store_id,
+            "transformationfunctions",
+        ]
+        headers = {"content-type": "application/json"}
+        return transformation_function.TransformationFunction.from_response_json(
+            _client._send_request(
+                "POST",
+                path_params,
+                headers=headers,
+                data=transformation_function_instance.json(),
+            )
+        )
+
+    def get_transformation_fn(self, name, version):
+        """
+        Retrieve transformation function from backend
+        Args:
+        name: TransformationFunction name, required
+            name of transformation function.
+        version: TransformationFunction version, required
+            version of transformation function.
+        """
+        _client = client.get_instance()
+        path_params = [
+            "project",
+            _client._project_id,
+            "featurestores",
+            self._feature_store_id,
+            "transformationfunctions",
+        ]
+
+        if name:
+            query_params = {"name": name}
+            if version:
+                query_params["version"] = version
+            return transformation_function.TransformationFunction.from_response_json(
+                _client._send_request("GET", path_params, query_params)
+            )
+        else:
+            return transformation_function.TransformationFunction.from_response_json(
+                _client._send_request("GET", path_params)
+            )
+
+    def delete(self, transformation_function_instance):
+        """Delete a transformation function.
+        Args:
+        transformation_function_instance: TransformationFunction, required
+            metadata object of transformation function.
+        """
+        _client = client.get_instance()
+        path_params = [
+            "project",
+            _client._project_id,
+            "featurestores",
+            self._feature_store_id,
+            "transformationfunctions",
+            transformation_function_instance.id,
+        ]
+        headers = {"content-type": "application/json"}
+        _client._send_request("DELETE", path_params, headers=headers)
+
+    def get_training_dataset_transformation_fn(self, training_dataset_instance):
```

Comment:
Why do we still have these long methods names?

Reply:
there is `get_transformation_fn` already but I will rename it as `get_td_transformation_fn`